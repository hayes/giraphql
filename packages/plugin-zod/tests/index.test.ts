import { gql } from 'apollo-server';
import { execute } from 'graphql';
import schema from './example/schema';

describe('mocked', () => {
  test('valid query', async () => {
    const query = gql`
      query {
        requiresOddArg(odd: 1)
      }
    `;

    const result = await execute({
      schema,
      document: query,
      contextValue: {},
    });

    expect(result).toMatchSnapshot();
  });

  test('invalid query', async () => {
    const query = gql`
      query {
        requiresOddArg(odd: 2)
      }
    `;

    const result = await execute({
      schema,
      document: query,
      contextValue: {},
    });

    expect(result).toMatchSnapshot();
  });
});
