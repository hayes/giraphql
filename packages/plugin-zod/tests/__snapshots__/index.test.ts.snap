// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mocked invalid query 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: 1 validation issue(s)

  Issue #0: custom_error at odd
  Number is not odd
],
  ],
}
`;

exports[`mocked valid query 1`] = `
Object {
  "data": Object {
    "requiresOddArg": 1,
  },
}
`;
